src/utlist.h

```diff
@@ -600,6 +620,31 @@ do {
   }                                                                                            \
 } while (0)
 
+#define DL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    DL_FIND_POS(_pos,add,cmp);                                                                 \
+    DL_APPEND_ELEM(head,_pos,add);                                                             \
+  } while (0)
+
+
+#define DL_FIND_POS(head,node,cmp)                                                             \
+  DL_FIND_POS2(head,node,cmp,prev,next)                                                        \
+
+#define DL_FIND_POS2(head,node,cmp,prev,next)                                                  \
+  do {                                                                                         \
+    if (cmp((head), (node)) > 0) {                                                             \
+      (head) = NULL;                                                                           \
+    } else {                                                                                   \
+      for (; (head)->next != NULL; head = (head)->next ) {                                     \
+        if ((head)->next == (node)) break;                                                     \
+        if (cmp((head)->next, (node)) > 0) {                                                   \
```

Comment:
Nit: `(node)` should be `node`, per [the above explanation](https://github.com/troydhanson/uthash/pull/101/files#r96501393).